workflows/energyplus.py

```diff
@@ -203,281 +190,306 @@ def run_energyplus(self, isIP, run_directory, file_name, args, by_api):
+            if platform.system() == 'Windows':
+                expand_objects = os.path.join(energyplus_root_folder, 'ExpandObjects.exe')
             else:
-                # if by CLI then just execute the full command line
-                try:
-                    for message in self.execute_for_callback(command_line_args, run_directory):
-                        self.callback(message)
-                except subprocess.CalledProcessError:
-                    self.callback("E+ FAILED")
-                    return EPLaunchWorkflowResponse1(
-                        success=False,
-                        message="EnergyPlus failed for file: %s!" % full_file_path,
-                        column_data={}
-                    )
+                expand_objects = os.path.join(energyplus_root_folder, 'ExpandObjects')
+            if not os.path.exists(expand_objects):
+                return EPLaunchWorkflowResponse1(
+                    success=False,
+                    message="ExpandObjects binary not found: {}!".format(expand_objects),
+                    column_data=[]
+                )
+            # go ahead and copy the target IDF and original IDD into the output_dir/in.idf for running ExpandObjects
+            idf_path_in_output_dir = os.path.join(output_directory, 'in.idf')
+            shutil.copy(full_file_path, idf_path_in_output_dir)
+            idd_path = os.path.join(energyplus_root_folder, 'Energy+.idd')  # yes for now we still need it
+            idd_path_in_output_dir = os.path.join(output_directory, 'Energy+.idd')
+            shutil.copy(idd_path, idd_path_in_output_dir)
+            # run ExpandObjects
+            try:
+                for message in self.execute_for_callback([expand_objects], output_directory):
+                    self.callback(message)
+            except subprocess.CalledProcessError:
+                self.callback("E+ FAILED")
+                return EPLaunchWorkflowResponse1(
+                    success=False,
+                    message="EnergyPlus failed for file: %s!" % full_file_path,
+                    column_data={}
+                )
+            # regardless of what happened, clean up the temp files
+            delete_if_exists(idf_path_in_output_dir)
+            delete_if_exists(idd_path_in_output_dir)
+            # if expanded.idf does not exist then we don't need to do anything, assume it just didn't have any templates
+            # but if it does exist, then this needs to be the IDF that we run, *however* we can't just leave it here
+            # as it may reference some other files back in the original directory.  I'm going to go with a new file name
+            # in the original directory of "FileName_Expanded.idf"
+            expanded_path_in_output_dir = os.path.join(output_directory, 'expanded.idf')
+            if os.path.exists(expanded_path_in_output_dir):
+                target_expanded_file_name = os.path.join(run_directory, f"{file_name_no_ext}_Expanded.idf")
```

Comment:
Here is the interesting part.  If the IDF is indeed expanded, it will become `<original_idf_name>_Expanded.idf` in the original idf directory.  So you'll have `HVACFile.idf` and `HVACFile_Expanded.idf`.  I am certainly not stuck on this behavior @JasonGlazer, but I **did** need a way to put the expanded IDF **back** into the original IDF directory.  This is a necessary step because there may be any number of supporting files (Python...) that also live right next to the original IDF and we want to leverage the benefits of running from the original location.  

With this current approach, the output files will have an extra `_Expanded` added on to them.  I guess that's OK...?  Gets a little weird for sure.  We'll need to discuss, @JasonGlazer 

I think there is one alternative reality where EnergyPlus itself expands the templates on the fly, and while in the past I have balked at that, I'm starting to think it would have more pros than cons.